scene/3d/sprite_3d.cpp

```diff
@@ -767,11 +822,12 @@ bool Sprite3D::is_region_enabled() const {
 }
 
 void Sprite3D::set_region_rect(const Rect2 &p_region_rect) {
-	bool changed = region_rect != p_region_rect;
-	region_rect = p_region_rect;
-	if (region && changed) {
-		_queue_redraw();
+	if (region_rect == p_region_rect) {
+		return;
 	}
+
+	region_rect = p_region_rect;
+	_queue_redraw();
```

Comment:
We can still keep the optimization and avoid redrawing if `region` is not enabled.